server/routes/createGoal/createGoalController.ts

```diff
@@ -75,17 +76,26 @@ export default class CreateGoalController {
     const targetDate = parseDate(req, 'targetDate')
     req.session.addStepForm = { ...req.body, targetDate }
     const { addStepForm } = req.session
+    const { addStepForms } = req.session
```

Comment:
Minor, and I'm not going to lose any sleep over this as its more weird TS syntax !!!, but because the 2 lines above are destructuring from the same variable, I think you can combine them into 1 statement, like this:
```
const { addStepForm, addStepForms } = req.session
```